### ISSUE TITLE
React prep

### ISSUE THREAD

**Issue by colevandersWands** [HackYourFutureBelgium/improvements#6]
How can we better prepare students for React before they reach the module?  As things are there are too many new concepts that are introduced at the same time as React, students are finding it hard to distinguish what's React, what's JS, and what's good practice.  

I think building two lists would be helpful, one for observed misunderstandings or missing knowledge and the other for ideas of how to introduce these concepts before they reach React:

_things that could be taught before to help when they reach React_:
- [ ] Class-based inheritance
- [ ] Dependency injection 
- [ ] Build scripts and transpiling
- [ ] JSX
- [ ] ?

_ways to incorporate these concepts earlier on_:
- [ ] ?

**Comment by KevinTss** [comment]
Probably first see the concept of components? 
It's something that we can easily implement in vanilla also like the DRY pattern

**Comment by colevandersWands** [comment]
@KevinTss, yup.  There's [this tutorial](https://watchandcode.com/p/practical-javascript) that introduces them which we assigned for class 6 and it was a big hit, then there's [these exercises](https://github.com/be-hacking-hyf/javascript-3/tree/master/dom-components).  if we do these in the first module where they learn OOP they'll hopefully be ready for classes that return components by the third week.

What did you have in mind for introducing DRY pattern?

**Comment by KevinTss** [comment]
> What did you have in mind for introducing DRY pattern?

Yup exactly that
